AttributeError: 'Attribute' object has no attribute 'name'

### Bug description



Minimal example causing crash:
```
class Item:
    def __init__(self, parent):
        self.__parent = parent
        self.__item = self.__parent.__item
```

Code more similar to my real use case, maybe it will add some context:
```
from __future__ import annotations

from typing import Optional


class Item:
    def __init__(self, parent: Optional[Item] = None):
        self.__parent = parent
        self.__item = self.__parent.__item if parent is not None else 'default'
```

### Configuration

_No response_

### Command used

```shell
echo -e "class Item:\n    def __init__(self, parent):\n        self.__parent = parent\n        self.__item = self.__parent.__item" > reproduce_error.py
pylint reproduce_error.py
```


### Pylint output

```shell
************* Module reproduce_error
reproduce_error.py:1:0: C0114: Missing module docstring (missing-module-docstring)
reproduce_error.py:1:0: C0115: Missing class docstring (missing-class-docstring)
reproduce_error.py:4:22: W0212: Access to a protected member __item of a client class (protected-access)
Exception on node <ClassDef.Item l.1 at 0x7f3880eb0730> in file '/home/dev/Desktop/pylint-error/reproduce_error.py'
Traceback (most recent call last):
  File "/home/dev/Desktop/pylint-error/venv/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 78, in walk
    callback(astroid)
  File "/home/dev/Desktop/pylint-error/venv/lib/python3.8/site-packages/pylint/checkers/classes/class_checker.py", line 877, in leave_classdef
    self._check_unused_private_attributes(node)
  File "/home/dev/Desktop/pylint-error/venv/lib/python3.8/site-packages/pylint/checkers/classes/class_checker.py", line 996, in _check_unused_private_attributes
    and attribute.expr.name == "self"
AttributeError: 'Attribute' object has no attribute 'name'
reproduce_error.py:1:0: F0001: Fatal error while checking 'reproduce_error.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/dev/.cache/pylint/pylint-crash-2022-05-26-18.txt'. (fatal)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```


### Expected behavior

No crash.

### Pylint version

```shell
pylint 2.13.9
astroid 2.11.5
Python 3.8.13 (default, Apr 19 2022, 00:53:22) 
[GCC 7.5.0]
```


### OS / Environment

Ubuntu 20.04.4 LTS

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -79,6 +79,10 @@ Release date: TBA
 
   Closes #6372
 
+* Fixed a crash in the ``unused-private-member`` checker involving chained private attributes.
+
+  Closes #6709
+
 * Fixed failure to enable ``deprecated-module`` after a ``disable=all``
   by making ``ImportsChecker`` solely responsible for emitting ``deprecated-module`` instead
   of sharing responsibility with ``StdlibChecker``. (This could have led to double messages.)

--- a/doc/whatsnew/2.14.rst
+++ b/doc/whatsnew/2.14.rst
@@ -165,6 +165,10 @@ Other Changes
 
   Closes #6372
 
+* Fixed a crash in the ``unused-private-member`` checker involving chained private attributes.
+
+  Closes #6709
+
 * Disable spellchecking of mypy rule names in ignore directives.
 
   Closes #5929

--- a/pylint/checkers/classes/class_checker.py
+++ b/pylint/checkers/classes/class_checker.py
@@ -1028,7 +1028,7 @@ a metaclass class method.",
                 if attribute.attrname != assign_attr.attrname:
                     continue
 
-                if isinstance(attribute.expr, nodes.Call):
+                if not isinstance(attribute.expr, nodes.Name):
                     continue
 
                 if assign_attr.expr.name in {

--- a/tests/functional/u/unused/unused_private_member.py
+++ b/tests/functional/u/unused/unused_private_member.py
@@ -344,3 +344,10 @@ class TypeSelfCallInMethod:
 
     def a(self):
         return type(self).__a
+
+
+class Item:
+    """Regression test for https://github.com/PyCQA/pylint/issues/6709"""
+    def __init__(self, parent):
+        self.__parent: Item = parent
+        self.__item = self.__parent.__item  # [unused-private-member]

--- a/tests/functional/u/unused/unused_private_member.txt
+++ b/tests/functional/u/unused/unused_private_member.txt
@@ -18,3 +18,4 @@ unused-private-member:276:4:276:31:FalsePositive4849.__unused_private_method:Unu
 unused-private-member:293:4:293:23:Pony.__init_defaults:Unused private member `Pony.__init_defaults(self)`:UNDEFINED
 unused-private-member:298:4:298:23:Pony.__get_fur_color:Unused private member `Pony.__get_fur_color(self)`:UNDEFINED
 unused-private-member:343:8:343:15:TypeSelfCallInMethod.b:Unused private member `TypeSelfCallInMethod.__a`:UNDEFINED
+unused-private-member:353:8:353:19:Item.__init__:Unused private member `Item.__item`:UNDEFINED

